rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -240,9 +241,9 @@ rclcpp::remove_ros_arguments(int argc, char const * const argv[])
   }
 
   std::vector<std::string> return_arguments;
-  return_arguments.resize(nonros_argc);
+  return_arguments.resize(static_cast<uint32_t>(nonros_argc));
 
-  for (int ii = 0; ii < nonros_argc; ++ii) {
+  for (uint32_t ii = 0U; ii < static_cast<uint32_t>(nonros_argc); ++ii) {
```

Comment:
If we are casting `nonros_argc` to a number twice, we should probably just store it above and reuse it rather than continually casting.  This may also change to `size_t` based on the previous comment.